whois-api/src/test/java/net/ripe/db/whois/api/rest/WhoisRestServiceTestIntegration.java

```diff
@@ -1835,6 +1835,67 @@ public void create_succeeds() {
         assertThat(whoisResources.getTermsAndConditions().getHref(), is(WhoisResources.TERMS_AND_CONDITIONS));
     }
 
+
+
+    @Test
+    public void create_asSet_non_auth_aut_num_succeeds() {
+        final RpslObject TEST_AS_SET = RpslObject.parse("""
+                as-set:     AS3333:AS-TEST
```

Comment:
Can we add test case that for flat hierarchy it does not change the source?

Reply:
if you try to create a flat as-set you will receive an error and if you try to update and you are updating the source you will receive an error as well